### PR TITLE
refactor(app-start,fx-firstrun-v1): move HeightObserver

### LINKED ISSUE #4719: Try to move the HeightObserver from app-start to fx-firstrun auth_broker
Ref: https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/lib/app-start.js has the heightobserver, which we might be able to clean up a bit.

### PR DESCRIPTION
**Pull request by himanish-star**
I hope this fixes #4719 .

@seanmonstar , I have also put the HeightObserver code into constructor as you had desired

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 93ccd996b6b5c0240ae1991114f1b99e6a81f83e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by vladikoff** [comment]
@himanish-star there are some failing tests that you can see running by going to : https://127.0.0.1:3030/tests/index.html

```
Failed test: returns a FxFirstrunV1 broker if `service=sync&context=iframe`
 => this._isInAnIframe is not a function
constructor@http://127.0.0.1:3030/tests/index.html?coverage&travis=true:190:5
initializeAuthenticationBroker@http://127.0.0.1:3030/tests/index.html?coverage&travis=true:1548:30
testExpectedBrokerCreated@http://127.0.0.1:3030/tests/spec/lib/app-start.js:253:16
@http://127.0.0.1:3030/tests/spec/lib/app-start.js:283:18
callFn@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4338:18
Runnable.prototype.run@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4331:7
Runner.prototype.runTest@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4728:5
next/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4806:7
next@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4653:14
next/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4663:7
next@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4601:23
Runner.prototype.hook/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4630:5
timeslice@http://127.0.0.1:3030/bower_components/mocha/mocha.js:5761:5
Failed test: returns a FxFirstrunV2 broker if `service=sync&context=fx_firstrun_v2`
 => this._isInAnIframe is not a function
constructor@http://127.0.0.1:3030/tests/index.html?coverage&travis=true:190:5
child@http://127.0.0.1:3030/bower_components/backbone/backbone.js:1567:34
initializeAuthenticationBroker@http://127.0.0.1:3030/tests/index.html?coverage&travis=true:1548:30
testExpectedBrokerCreated@http://127.0.0.1:3030/tests/spec/lib/app-start.js:253:16
@http://127.0.0.1:3030/tests/spec/lib/app-start.js:294:18
callFn@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4338:18
Runnable.prototype.run@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4331:7
Runner.prototype.runTest@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4728:5
next/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4806:7
next@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4653:14
next/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4663:7
next@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4601:23
Runner.prototype.hook/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4630:5
timeslice@http://127.0.0.1:3030/bower_components/mocha/mocha.js:5761:5
Failed test: sets up the HeightObserver, triggers a `resize` notification on the iframe channel when the height changes
 => appStart.initializeHeightObserver is not a function
@http://127.0.0.1:3030/tests/spec/lib/app-start.js:647:9
Runnable.prototype.run@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4311:7
Runner.prototype.runTest@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4728:5
next/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4806:7
next@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4653:14
next/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4663:7
next@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4601:23
next/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4625:7
done@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4300:5
callFn@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4343:7
Runnable.prototype.run@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4331:7
next@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4613:5
Runner.prototype.hook/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4630:5
timeslice@http://127.0.0.1:3030/bower_components/mocha/mocha.js:5761:5
Failed test: "before each" hook
 => this._isInAnIframe is not a function
constructor@http://127.0.0.1:3030/tests/index.html?coverage&travis=true:190:5
@http://127.0.0.1:3030/tests/spec/models/auth_brokers/fx-firstrun-v1.js:42:16
callFn@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4338:18
Runnable.prototype.run@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4331:7
next@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4613:5
Runner.prototype.hook/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4630:5
timeslice@http://127.0.0.1:3030/bower_components/mocha/mocha.js:5761:5
Failed test: "before each" hook
 => this._isInAnIframe is not a function
constructor@http://127.0.0.1:3030/tests/index.html?coverage&travis=true:190:5
child@http://127.0.0.1:3030/bower_components/backbone/backbone.js:1567:34
@http://127.0.0.1:3030/tests/spec/models/auth_brokers/fx-firstrun-v2.js:28:16
callFn@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4338:18
Runnable.prototype.run@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4331:7
next@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4613:5
Runner.prototype.hook/<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4630:5
timeslice@http://127.0.0.1:3030/bower_components/mocha/mocha.js:5761:5
× firefox on any platform - mocha tests - run the mocha tests (56.703s)

```

File: app/scripts/models/auth_brokers/fx-firstrun-v1.js [hunk 1]
```diff
@@ -26,6 +27,21 @@ define(function (require, exports, module) {
       VERIFICATION_COMPLETE: 'verification_complete'
     },
 
+    constructor () {
+      if (this._isInAnIframe()) {
```
Comment (vladikoff) [inline app/scripts/models/auth_brokers/fx-firstrun-v1.js]: not sure if this function is available here, you might need to use the options passed into the `auth_broker` instead to determine this

Comment (himanish-star) [reply app/scripts/models/auth_brokers/fx-firstrun-v1.js]: @vladikoff , thanks for suggestions working on them now

**Comment by vladikoff** [comment]
CI was still failing, please submit a new rebased PR to merge this

**Comment by himanish-star** [comment]
Are the changes correct? Or do I have to make some changes?

File: app/scripts/models/auth_brokers/fx-firstrun-v1.js [hunk 2]
```diff
@@ -26,6 +28,25 @@ define(function (require, exports, module) {
       VERIFICATION_COMPLETE: 'verification_complete'
     },
 
+    _isInAnIframe () {
+      return new Environment(window).isFramed();
```
Comment (vladikoff) [inline app/scripts/models/auth_brokers/fx-firstrun-v1.js]: check if this needs to be `this._window`

Also tests need to be adjusted from app-start into this file

Comment (himanish-star) [reply app/scripts/models/auth_brokers/fx-firstrun-v1.js]: ok :+1:

**Pull request closed without merging**